sat/apiclient/fas.py

```diff
@@ -339,13 +353,21 @@ def _create_temporary_snapshot(self, xnames=None):
         # retrieve it and poll until ready
         response = None
         ready = False
-        while not ready:
-            time.sleep(1)
+        consecutive_failures = 0
+        while not ready and consecutive_failures < 5:
```

Comment:
It looks like the check on `consecutive_failures` is unnecessary because it's handled any time it's incremented on line 365 below.

Reply:
Got it. Removed on line 357.